### PR TITLE
Fix unavailable hvac-status on Zoe50

### PR DESCRIPTION
**Pull request by ZMan88**
It now seems to work for Zoe Phase 2

### CODE REVIEW

**Comment by epenet** [comment]
Do you have some sample results?
Maybe you could add a positive fixture like this, as we only seem to have a negative one currently :
https://github.com/hacf-fr/renault-api/blob/56ad77f7ebd27a567d6a16303a5c17bb7e3f7729/tests/fixtures/kamereon/vehicle_data/hvac-status.json#L1-L7

**Comment by ZMan88** [comment]
Ok, 
I will add that, but the response I get from the api is slightly different, it looks like this:
{
    "id": "xxx MY VIN xxx",
    "socThreshold": 40,
    "lastUpdateTime": "2021-12-27T09:37:39Z",
    "hvacStatus": "off"
}
the format that you are expecting is from where?
Thanks, and sorry for the clumsy approach :)

**Comment by epenet** [comment]
This is still an "off" status. Do you a real fixture with "on" status or does it always show "off" anyway?
On Zoe40, it shows "off" even when hvac is "on".

**Comment by ZMan88** [comment]
It shows on when the preconditioning is triggered. Does it need to be set to "on" in the fixture?

**Comment by epenet** [comment]
Yes, we need to make sure that we have a "real positive" in the fixture.

**Comment by epenet** [comment]
Are you able to run pytest locally before you push commits?
Unless all tests pass I cannot merge...

**Comment by ZMan88** [comment]
I am having some trouble with my dev environment. I will try to fix it and run the tests locally and then push changes

**Comment by epenet** [comment]
Thanks

File: tests/fixtures/kamereon/vehicle_data/hvac-status.zoe_50.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "data": {
+    "type": "Car",
+    "id": "VF1AAAAA555777999",
+    "attributes": {
+      "socThreshold": 40,
+      "hvacStatus": "on"
+    }
+  }
+}
```
Comment (epenet) [inline tests/fixtures/kamereon/vehicle_data/hvac-status.zoe_50.json:10]: Please add the missing carriage return here:
```suggestion
}

```

File: tests/kamereon/test_kamereon_vehicles.py [hunk 2]
```diff
@@ -125,9 +125,7 @@ def test_vehicles_response(filename: str) -> None:
                 "reports_charging_power_in_watts": vehicle_details.reports_charging_power_in_watts(),  # noqa: B950
                 "uses_electricity": vehicle_details.uses_electricity(),
                 "uses_fuel": vehicle_details.uses_fuel(),
-                "supports-hvac-status": vehicle_details.supports_endpoint(
-                    "hvac-status"
-                ),
+                "supports-hvac-status": vehicle_details.supports_endpoint("hvac-status"),
```
Comment (epenet) [inline tests/kamereon/test_kamereon_vehicles.py:128]: Please revert this change, it is causing formatting issues.
```suggestion
                "supports-hvac-status": vehicle_details.supports_endpoint(
                    "hvac-status"
                ),
```

**Review by epenet — COMMENTED** [review]
Please fix these formatting issues

**Comment by epenet** [comment]
To fix the tests, please ensure that you update the expected output for `zoe_50.1.json` in `tests/cli/test_vehicle.py` (line 100):
```python
    "zoe_50.1.json": (
        "--------------------  -------------------------\n"
        "Battery level         50 %\n"
        "Last updated          2020-11-17 09:06:48\n"
        "Range estimate        128 km\n"
        "Plug state            PlugState.UNPLUGGED\n"
        "Charging state        ChargeState.NOT_IN_CHARGE\n"
        "Charge mode           always\n"
        "Total mileage         49114.27 km\n"
        "GPS Latitude          48.1234567\n"
        "GPS Longitude         11.1234567\n"
        "GPS last updated      2020-02-18 17:58:38\n"
        "HVAC status           off\n"
        "External temperature  8.0 °C\n"
        "--------------------  -------------------------\n"
    ),
```

File: tests/cli/test_vehicle.py [hunk 3]
```diff
@@ -108,6 +108,7 @@
         "Total mileage     49114.27 km\n"
         "GPS Latitude      48.1234567\n"
         "GPS Longitude     11.1234567\n"
+        "HVAC status           on\n"
```
Comment (epenet) [inline tests/cli/test_vehicle.py:111]: This still has spacing issues. All the text needs to be aligned.

**Comment by epenet** [comment]
Please review the formatting issues from https://github.com/hacf-fr/renault-api/runs/4672946801?check_suite_focus=true
And please adjust the tests a little more - GPS should be before HVAC status: https://github.com/hacf-fr/renault-api/runs/4672947004?check_suite_focus=true

**Comment by epenet** [comment]
We're almost there - there are still some formatting issues.
Could you please adjust?

**Comment by ZMan88** [comment]
I will love to but unfortunately I can't figure out what the problems are...
Any pointers?
Also what IDE are u using? I am using VS Code.

**Comment by epenet** [comment]
If you look at the test details, you will see it simply a missing comma here or there: https://github.com/hacf-fr/renault-api/runs/4685458157?check_suite_focus=true

**Comment by ZMan88** [comment]
I'm sorry but I cannot figure out what's wrong 🙈

File: src/renault_api/kamereon/models.py [hunk 4]
```diff
@@ -348,6 +347,7 @@ class KamereonVehicleLocationData(KamereonVehicleDataAttributes):
 class KamereonVehicleHvacStatusData(KamereonVehicleDataAttributes):
     """Kamereon vehicle data hvac-status attributes."""
 
+    socThreshold: Optional[float]
```
Comment (epenet) [inline src/renault_api/kamereon/models.py:350]:
```suggestion
    socThreshold: Optional[float]  # noqa: N815
```

File: tests/fixtures.py [hunk 5]
```diff
@@ -409,6 +414,6 @@ def inject_vehicle_status(mocked_responses: aioresponses, vehicle: str) -> None:
     """Inject Kamereon vehicle status data."""
     inject_get_battery_status(mocked_responses)
     inject_get_location(mocked_responses)
-    inject_get_hvac_status(mocked_responses)
+    inject_get_hvac_status(mocked_responses,vehicle)
```
Comment (epenet) [inline tests/fixtures.py:417]:
```suggestion
    inject_get_hvac_status(mocked_responses, vehicle)
```

File: tests/test_renault_vehicle.py [hunk 6]
```diff
@@ -119,11 +119,19 @@ async def test_get_location(
 
 
 @pytest.mark.asyncio
-async def test_get_hvac_status(
+async def test_get_hvac_status_zoe(
     vehicle: RenaultVehicle, mocked_responses: aioresponses
 ) -> None:
     """Test get_hvac_status."""
-    fixtures.inject_get_hvac_status(mocked_responses)
+    fixtures.inject_get_hvac_status(mocked_responses,"zoe")
```
Comment (epenet) [inline tests/test_renault_vehicle.py:126]:
```suggestion
    fixtures.inject_get_hvac_status(mocked_responses, "zoe")
```

File: tests/test_renault_vehicle.py [hunk 7]
```diff
@@ -119,11 +119,19 @@ async def test_get_location(
...
+    assert await vehicle.get_hvac_status()
+
+@pytest.mark.asyncio
+async def test_get_hvac_status_zoe50(
+    vehicle: RenaultVehicle, mocked_responses: aioresponses
+) -> None:
+    """Test get_hvac_status."""
+    fixtures.inject_get_hvac_status(mocked_responses,"zoe_50")
```
Comment (epenet) [inline tests/test_renault_vehicle.py:134]:
```suggestion
    fixtures.inject_get_hvac_status(mocked_responses, "zoe_50")
```

**Review by epenet — CHANGES_REQUESTED** [review]
Please fix the pre-commit warnings

**Comment by epenet** [comment]
You should be able to run `pre-commit run --all-files` to fix the format issues.

File: tests/fixtures.py [hunk 8]
```diff
@@ -235,13 +235,16 @@ def inject_get_location(mocked_responses: aioresponses) -> str:
     )
 
 
-def inject_get_hvac_status(mocked_responses: aioresponses) -> str:
+def inject_get_hvac_status(mocked_responses: aioresponses, vehicle: str) -> str:
     """Inject sample hvac-status."""
     urlpath = f"{ADAPTER_PATH}/hvac-status?{DEFAULT_QUERY_STRING}"
+    filename = "hvac-status.json"
+    if "zoe" in vehicle:
+        filename = f"hvac-status.{vehicle}.json"
```
Comment (epenet) [inline tests/fixtures.py:243]:
```suggestion
    filename = "hvac-status.1.json"
    if vehicle in "zoe_50":
        filename = f"hvac-status.2.json"
```

**Pull request merged**